src/NATS.Client.Core/INatsConnection.cs

```diff
@@ -154,4 +171,11 @@ public interface INatsConnection
         NatsPubOpts? requestOpts = default,
         NatsSubOpts? replyOpts = default,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Connect socket and write CONNECT command to nats server.
+    /// </summary>
+    ValueTask ConnectAsync();
+
+    NatsStats GetStats();
```

Comment:
We have been quite careful about which methods we make public on the interface. Can we focus on the ones you need rather than extracting every public method into the interface?

Reply:
@mtmk We probably don't need this one, but it is public on the class.   We don't know every method and property we will need yet since we can't yet use the new library due to missing interface injection. For interfaces, they should be swappable for the object instance so if the instance has the method public, it should be in the interface.

Reply:
That's fair enough I should've been open about my motive here; I think we should start using .NET `Metrics` for Open telemetry. could we make `GetSTats()` internal as well?